.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - staging
+      - trying
```

Comment:
After the `push:` event, you can add the `workflow_dispatch:` event to enable manually triggering the workflow for spurious errors.

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch
https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

Reply:
Oh, that could be useful. Maybe for releases... Thanks for letting me know! And sorry if I spammed your inbox.

Reply:
I'm going to add more/better checks later, but I think I have had enough of this for today. It would be nice to check formatting and clippy, for example. This was more of an emergency migration since it would have eaten all my credits in no time.

Reply:
No worries. I had the same issue of migrating and having to learn/piece together workflows because the free credits ran out.

<details>
  <summary>This is what I have for clippy and rustfmt, it won't fail a build on warning but does the checks anyway.</summary>

```yaml
  clippy-rustfmt:
    name: Clippy and rustfmt
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions-rs/toolchain@v1
        with:
          profile: minimal
          toolchain: stable
          components: clippy, rustfmt
          override: true
      - name: clippy
        run: cargo clippy
        continue-on-error: true
      - name: rustfmt
        run: cargo fmt -- --check
        continue-on-error: true
```
</details

Reply:
That looks like a good setup. I should do something similar.